addons/survey/models/survey_survey.py

```diff
@@ -365,6 +367,15 @@ def _compute_session_available(self):
         for survey in self:
             survey.session_available = survey.survey_type in {'live_session', 'custom'}
 
+    @api.depends_context('uid')
+    def _compute_allowed_survey_types(self):
+        self.allowed_survey_types = ([
+            ('survey', 'Survey'),
+            ('live_session', 'Live session'),
+            ('assessment', 'Assessment'),
+            ('custom', 'Custom'),
+        ] if self.env.user.has_group('survey.group_survey_user') else [])
```

Comment:
If `[]` is stored as `False`anyway, let's simplify
```suggestion
        self.allowed_survey_types = [
            ('survey', 'Survey'),
            ('live_session', 'Live session'),
            ('assessment', 'Assessment'),
            ('custom', 'Custom'),
        ] if self.env.user.has_group('survey.group_survey_user') else False
```